### PR TITLE
dunfell/rz sbc:  enhance layer handling and update documentation

### PR DESCRIPTION
**Pull request by tiennguyenuh**
Hi @preetam-reddy, @vudangRVC,

Change points:
  - Added a new function to retry git clone up to 5 times if it fails, with error handling for persistent failures.
  - Added a new function to download missing layers in `yocto_rzsbc_board`.
  - Added a new function to verify tags, commits and branchs of meta layers before the build.
  - Added repository configuration field to the JSON file.
  - Updated the location of the README.md file to be deployed to the host/src folder.
  
In this PR, the structure of `git_patch.json` has been changed to include additional fields for repository configuration. Should we rename the file as well? If we rename it, we will also need to update the documentation in the build script and meta layer.

### CODE REVIEW

File: rz-sbc/README.md [hunk 1]
```diff
@@ -22,11 +23,33 @@ This directory holds the automated build scripts that perform the rz yocto repos
 
 | File           | Description                                                                                                                                       |
 |----------------|---------------------------------------------------------------------------------------------------------------------------------------------------|
-| git_patch.json | contains json keys and list of patchfiles to apply.                                                                                               |
+| git_patch.json | contains json keys and repository configuration such as: url, branch, tag, commit and patch paths to apply. type.                                                                                               |
```
Comment (preetam-reddy) [inline rz-sbc/README.md]: type?

File: rz-sbc/README.md [hunk 2]
```diff
@@ -46,11 +69,11 @@ To perform yocto build with all RZ SOc's IP's functioning, you will need to down
 > Please ensure that you are making this build in an ubuntu 20.04 OS environment through docker/VM/native-OS installations.
 
 Once you download the packages, place the zip files here.
-Then rerun the build script and it will take ov everythig else.
+Then rerun the build script and it will take of everything else.
```
Comment (preetam-reddy) [inline rz-sbc/README.md]: > take care of everything

File: rz-sbc/rzsbc_yocto.sh [hunk 3]
```diff
@@ -52,36 +49,67 @@ guideline() {
     echo "------------------------------------------------------------"
 }
 
+parse_json_repo() {
+    # Initialize arrays as global variables
+    declare -g -A repo_urls
+    declare -g -A repo_branches
+    declare -g -A repo_commits
+    declare -g -A repo_tags
+    declare -g -A repo_types
```
Comment (preetam-reddy) [inline rz-sbc/rzsbc_yocto.sh]: Is this posix compliant?

File: rz-sbc/README.md (hunk 1 above)
Comment (tiennguyenuh) [reply rz-sbc/README.md]: Updated in the commit: https://github.com/Renesas-SST/rz-build-scripts/commit/92de55de36b303bf0048afe21f36c32050444cfb

File: rz-sbc/README.md (hunk 2 above)
Comment (tiennguyenuh) [reply rz-sbc/README.md]: Updated in the commit: https://github.com/Renesas-SST/rz-build-scripts/commit/92de55de36b303bf0048afe21f36c32050444cfb

File: rz-sbc/rzsbc_yocto.sh (hunk 3 above)
Comment (tiennguyenuh) [reply rz-sbc/rzsbc_yocto.sh]: Updated in the commit: https://github.com/Renesas-SST/rz-build-scripts/commit/5876cd9016ddbc1e19d76c213712e49ef0468071

File: rz-sbc/rzsbc_yocto.sh [hunk 4]
```diff
@@ -52,43 +49,51 @@ guideline() {
...
-        local_patch=$(echo "$local_patch" | xargs)
-        if [ -f "$TOP_DIR/$local_patch" ]; then
-            echo "Applying local patch: $TOP_DIR/$local_patch"
-            echo $PWD
+    echo "$patch_list" | while IFS= read -r local_patch; do
+    local_patch=$(echo "$local_patch" | xargs)
+    if [ -f "$TOP_DIR/$local_patch" ]; then
+        echo "Applying local patch: $TOP_DIR/$local_patch"
+        if [ ! -d ".git" ]; then
+            patch -p1 < "$TOP_DIR/$local_patch"
```
Comment (preetam-reddy) [inline rz-sbc/rzsbc_yocto.sh:87]: I like this. You are supporting regular patch as well as git format patch. Good work.

**Review by preetam-reddy — APPROVED** [review]

**Pull request merged**